model/ResultsService.php

```diff
@@ -1204,30 +1230,104 @@ private function getItemIndexer($delivery)
     {
         $deliveryUri = $delivery->getUri();
         if (!array_key_exists($deliveryUri, $this->indexerCache)) {
-            $runtime = $this->getServiceLocator()->get(AssignmentService::SERVICE_ID)->getRuntime($delivery);
-            $inputParameters = \tao_models_classes_service_ServiceCallHelper::getInputValues($runtime, []);
-            $directoryIds = explode('|', $inputParameters['QtiTestCompilation']);
-            $indexer = $this->getDecompiledIndexer($directoryIds[0]);
+            $directory = $this->getPrivateDirectory($delivery);
+            $indexer = $this->getDecompiledIndexer($directory);
             $this->indexerCache[$deliveryUri] = $indexer;
         }
         $indexer = $this->indexerCache[$deliveryUri] ;
         return $indexer;
     }
 
     /**
-     * @param string $directoryId
-     * @return QtiTestCompilerIndex
+     * @param $delivery
+     * @param $testUri
+     * @return ResourceCompiledMetadataHelper
      */
-    private function getDecompiledIndexer($directoryId)
+    private function getTestMetadata(core_kernel_classes_Resource $delivery, $testUri)
     {
-        $fileStorage = \tao_models_classes_service_FileStorage::singleton();
-        /** @var tao_models_classes_service_StorageDirectory $private */
-        $private = $fileStorage->getDirectoryById($directoryId);
+        if (array_key_exists($testUri, $this->testMetadataCache)) {
```

Comment:
Not maybe a big problem but I will use `isset` is faster and checks also if the value it's  null.
What do you think?

Reply:
Yes, it makes sense